exonum/src/storage/proof_map_index/proof.rs

```diff
@@ -653,13 +642,48 @@ where
         });
 
         collect(&proof).map(|h| {
-            (
-                entries
-                    .into_iter()
-                    .filter_map(OptionalEntry::into_kv)
-                    .collect(),
-                h,
-            )
+            CheckedMapProof {
+                entries: entries.into_iter().map(OptionalEntry::into).collect(),
+                hash: h,
+            }
         })
     }
 }
+
+impl<K, V> CheckedMapProof<K, V> {
```

Comment:
Would it make sense to add a consuming method returning `self.entries` and/or implement `IntoIterator<Item = (K, Option<V>)`?